hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -366,10 +392,19 @@ private void commitOnAutoCommit(String commitTime, JavaRDD<WriteStatus> resultRD
 
   private JavaRDD<HoodieRecord<T>> combineOnCondition(boolean condition, JavaRDD<HoodieRecord<T>> records,
       int parallelism) {
+    return condition ? deduplicateRecords(records, parallelism) : records;
+  }
+
+  private JavaRDD<HoodieKey> combineKeysOnCondition(boolean condition, JavaRDD<HoodieKey> keys,
```

Comment:
Again see if this separate function needed. See if this can be inline as it is used in only one place and looks trivial